src/Aspire.Hosting/MongoDB/MongoDbContainerResource.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a MongoDb container.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="password">The MongoDb root password.</param>
+public class MongoDbContainerResource(string name, string password) : ContainerResource(name), IMongoDbResource
+{
+    public string Password { get; } = password;
+
+    /// <summary>
+    /// Gets the connection string for the MongoDb server.
+    /// </summary>
+    /// <returns>A connection string for the MongoDb server in the form "mongodb://host:port".</returns>
+    public string? GetConnectionString()
+    {
+        if (!this.TryGetAllocatedEndPoints(out var allocatedEndpoints))
+        {
+            throw new DistributedApplicationException("Expected allocated endpoints!");
+        }
+
+        var allocatedEndpoint = allocatedEndpoints.Single();
+
+        return $"mongodb://root:{Password}@{allocatedEndpoint.Address}:{allocatedEndpoint.Port}";
```

Comment:
Does there need to be any escaping done on the password here?

Reply:
Hi @mitchdenny!

I've changed the implementation to deal with special chars, please review.